src/Commands/Invite.cpp

```diff
@@ -1,9 +1,65 @@
 #include "Commands/Invite.hpp"
+#include "AClient.hpp"
+#include "Server.hpp"
+#include "Channel.hpp"
 
 Invite::Invite( Server& ircServ ) : ICommand( ircServ ) {}
 
-Invite::~Invite( void ) {
+Invite::~Invite( void ) {}
 
+void Invite::initArgs( void ) {
+    const std::vector<std::string>& args = utils::split( _rawCommand, " " );
+    std::vector<std::string>::const_iterator it = args.begin();
+
+    _invitedNick = *( it++ );
+    _invitedClient = _ircServ._findClientByNick( _invitedNick );
+    std::map<const std::string, Channel*>::iterator chan_it = _ircServ._channels.find( *it );
+    if ( chan_it == _ircServ._channels.end() )
+        throw ( std::runtime_error( ERR_NOSUCHCHANNEL( _ircServ.getIp(), _client->getNick() + " " +  *it ) ) );
+    _channel = chan_it->second;
+}
+
+void Invite::validateArgs( void ) const {
+    if ( _rawCommand.find( "," ) != std::string::npos )
+        throw ( std::runtime_error( ERR_MULTIPLEPARAMS( _ircServ.getIp(), _client->getNick() + " INVITE " ) ) );
+
+    if ( !_channel->isMember( _client->getNick() ) )
+        throw ( std::runtime_error( ERR_NOTONCHANNEL( _ircServ.getIp(),
+            _client->getNick() + " " + _channel->getName() ) ) );
+    else if ( !_channel->isOperator( _client->getNick() ) && _channel->isInviteOnly() )
+        throw ( std::runtime_error( ERR_CHANOPRIVSNEEDED( _ircServ.getIp(),
+            _client->getNick() + " " + _channel->getName() ) ) );
+
+    if ( _invitedNick.empty() )
+        throw ( std::runtime_error( ERR_NEEDMOREPARAMS( _ircServ.getIp(), _client->getNick() + " INVITE " ) ) );
+    else if ( _channel->isMember( _invitedNick ) )
+        throw ( std::runtime_error( ERR_USERONCHANNEL( _ircServ.getIp(),
+            _client->getNick() + " " + _invitedNick + " " + _channel->getName() ) ) );
+    else if ( !_invitedClient )
```

Comment:
maybe check this first, since if there is no user with that nick on the server, why do we need to go through the rest of the checks.